runtime/environment.go

```diff
@@ -357,7 +359,37 @@ func (e *interpreterEnvironment) parseAndCheckProgram(
 	program *interpreter.Program,
 	err error,
 ) {
-	wrapError := func(err error) error {
+
+	// NOTE: Always ensure to call maybeSetProgram
+	// on all error paths of parseAndCheckProgram,
+	// except for once Interface.SetProgram was called!
+	//
+	// If the program is supposed to be stored,
+	// there was a call to Interface.GetProgram before.
+	// If an error occurs after calling Interface.GetProgram,
+	// then we *must* also call Interface.SetProgram, with nil.
+	//
+	// See the documentation for Interface.GetProgram
+	// for an explanation of this invariant.
+
+	maybeSetProgram := func() error {
+
+		if !storeProgram {
+			return nil
+		}
+
+		var err error
+		wrapPanic(func() {
+			err = e.runtimeInterface.SetProgram(location, program)
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
same as:
```suggestion
		return err
```

Reply:
Simplified in [f8f740d](https://github.com/onflow/cadence/pull/2241/commits/f8f740dd98f2012e3da0c5c70edd3d942d605073)